modules/olmv1-catalog-plain.adoc

```diff
@@ -0,0 +1,234 @@
+// Module included in the following assemblies:
+//
+// * operators/olm_v1/olmv1-managing-catalogs.adoc
+
+:_content-type: CONCEPT
+
+[id="olmv1-building-plain-bundle-image-source_{context}"]
+= Building a plain bundle image from an image source
+
+The Operator Controller currently supports installing plain bundles created only from a _plain bundle image_.
+
+.Prequisites
+
+* You have Kubernetes manifests for your bundle in a flat directory at the root of your project similar to the following structure:
++
+.Example directory structure
+[source,terminal]
+----
+manifests
+├── namespace.yaml
+├── service_account.yaml
+├── cluster_role.yaml
+├── cluster_role_binding.yaml
+└── deployment.yaml
+----
+
+.Procedure
+
+. At the root of your project, create a Dockerfile that can build a bundle image:
++
+.Example `plainbundle.Dockerfile`
+----
+FROM scratch <1>
+ADD manifests /manifests
+----
+<1> Use the `FROM scratch` directive to make the size of the image smaller. No other files or directories are required in the bundle image.
+
+. Build an OCI-compliant image using your preferred build tool, similar to the following example:
++
+[source,terminal]
+----
+$ podman build -f plainbundle.Dockerfile -t \
+    quay.io/<organization_name>/<repository_name>:<image_tag> . <1>
+----
+<1> Use an image tag that references a repository where you have push access privileges.
+
+. Push the image to your remote registry:
++
+[source,terminal]
+----
+$ podman push quay.io/<organization_name>/<repository_name>:<image_tag>
+----
+
+.Additional resources
+
+* link:https://github.com/operator-framework/olm-docs/blob/master/content/en/docs/Reference/file-based-catalogs.md[File-based catalog bundle schema]
+* link:https://github.com/opencontainers/image-spec#oci-image-format-specification[OCI image specification]
+* link:https://github.com/operator-framework/rukpak/blob/main/docs/bundles/plain.md#image-source[RukPak > Building a plain bundle > Image source]
+* link:https://github.com/operator-framework/rukpak/blob/main/docs/sources/image.md#private-image-registries[RukPak > Sources > Images > Private image registries]
+
+[id="olmv1-creating-fbc_{context}"]
+= Creating a file-based catalog
+
+If you do not have a file-based catalog, you must perform the following steps to initialize the catalog.
+
+.Procedure
+
+. Create a directory for the catalog by running the following command:
++
+[source,terminal]
+----
+$ mkdir <catalog_dir>
+----
+
+. In the same directory level, create a Dockerfile that can build a catalog image:
++
+[NOTE]
+====
+The Dockerfile must be in the same parent directory as the catalog directory that you created in the previous step:
+
+.Example directory structure
+[source,terminal]
+----
+.
+├── <catalog_dir>
+└── <catalog_dir>.Dockerfile
+----
+====
++
+.Example `<catalog_dir>.Dockerfile`
```

Comment:
This Dockerfile will not work since no `opm serve` in it. You can use the `opm generate dockerfile` to create one, such as
```console
[cloud-user@preserve-olm-env2 jian]$ opm generate dockerfile doc-catalog
[cloud-user@preserve-olm-env2 jian]$ cat doc-catalog.Dockerfile 
# The base image is expected to contain
# /bin/opm (with a serve subcommand) and /bin/grpc_health_probe
FROM quay.io/operator-framework/opm:latest

# Configure the entrypoint and command
ENTRYPOINT ["/bin/opm"]
CMD ["serve", "/configs", "--cache-dir=/tmp/cache"]

# Copy declarative config root into image at /configs and pre-populate serve cache
ADD doc-catalog /configs
RUN ["/bin/opm", "serve", "/configs", "--cache-dir=/tmp/cache", "--cache-only"]

# Set DC-specific label for the location of the DC root directory
# in the image
LABEL operators.operatorframework.io.index.configs.v1=/configs
```